framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -216,47 +184,124 @@ private static String replaceParameterSyntax(String expression) {
      * A visitor class that converts a JavaParser {@link Expression} to a {@link JavaExpression}.
      */
     private static class ExpressionToJavaExpressionVisitor
```

Comment:
This nested class is enormous, so we could consider moving it to its own file.  Leaving it here better encapsulates it, though.

Reply:
After this PR is merged. We should make a new package "org.checkerframework.framework.util.expression" and move JavaExpressionParseUtil and StringToJavaExpression to that package.  Also, ExpressionToJavaExpressionVisitor will be made a top level class in that package.